datashader/core.py

```diff
@@ -310,22 +298,34 @@ def raster(self,
         data : xarray.Dataset
 
         """
+        # For backwards compatibility
+        if agg         is None: agg=downsample_method
+        if interpolate is None: interpolate=upsample_method
+        
         upsample_methods = dict(nearest=US_NEAREST,
                                 linear=US_LINEAR)
 
-        downsample_methods = dict(first=DS_FIRST,
-                                  last=DS_LAST,
-                                  mean=DS_MEAN,
-                                  mode=DS_MODE,
-                                  var=DS_VAR,
-                                  std=DS_STD,
-                                  min=DS_MIN,
-                                  max=DS_MAX)
+        downsample_methods = {'first':DS_FIRST,
+                              'last':DS_LAST,
+                              'mode':DS_MODE,
+                              'mean':DS_MEAN,   rd.mean:DS_MEAN,
+                              'var':DS_VAR,     rd.var:DS_VAR,
+                              'std':DS_STD,     rd.std:DS_STD,
+                              'min':DS_MIN,     rd.min:DS_MIN,
+                              'max':DS_MAX,     rd.max:DS_MAX}
 
-        if upsample_method not in upsample_methods.keys():
+        if interpolate not in upsample_methods.keys():
             raise ValueError('Invalid upsample method: options include {}'.format(list(upsample_methods.keys())))
-        if downsample_method not in downsample_methods.keys():
+
+        if hasattr(agg,"column"):
+            # Accepts a reduction from ./reductions.py, but only uses the class name and the optional column name
+            if hasattr(source,"name") and agg.column != source.name:
```

Comment:
Did we decide that if agg.column is ``None`` it should match any name on the DataArray? Since it's unambiguous that seems fine to me.